fvm/evm/stdlib/contract.go

```diff
@@ -1528,7 +1528,7 @@ func newInternalEVMTypeDepositFunction(
 ) *interpreter.HostFunctionValue {
 	return interpreter.NewStaticHostFunctionValue(
 		gauge,
-		internalEVMTypeCallFunctionType,
+		internalEVMTypeDepositFunctionType,
```

Comment:
these are mistakes (wrong type passed) we made in the past but there was no complain by the Cadence. cc: @turbolent in case its uncovering a hidden bug on Cadence.

Reply:
Yeah, there is nothing that can be done here - if a function is implemented natively, like here, the developer is responsible for providing the correct type. The implementation of Cadence has nothing to check against

Reply:
got it, just want to make sure that's the correct behaviour, it seems this type that we passed was never enforced.